### PR TITLE
Build controller image before run e2e tests in Openshift CI

### PR DESCRIPTION
**Pull request by flacatus**
Signed-off-by: flacatus <flacatus@redhat.com>

### What does this PR do?
After merging this PR https://github.com/openshift/release/pull/10255 in Openshift CI now we are able to build controller image in Openshift CI and use the image in e2e tests. 
Also I delete the implementation of remove namespace after tests finishing because if tests will fail we can check the pod logs in PROW after finish the job.

### What issues does this PR fix or reference?
fixes https://github.com/eclipse/che/issues/17264

### CODE REVIEW

File: test/e2e/cmd/workspaces_test.go [hunk 1]
```diff
@@ -80,10 +80,6 @@ var _ = ginkgo.SynchronizedAfterSuite(func() {
 		_ = fmt.Errorf("Failed to uninstall workspace controller %s", err)
 	}
 
-	if err = k8sClient.Kube().CoreV1().Namespaces().Delete(config.Namespace, &metav1.DeleteOptions{}); err != nil {
```
[the remark this answers is not in the document]
Comment (flacatus) [reply test/e2e/cmd/workspaces_test.go:83]: The namespace will be deleted with AWS cluster. If we don't remove the namespace we can have all devworkspace logs here: https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/devfile_devworkspace-operator/137/pull-ci-devfile-devworkspace-operator-master-v5-openshift-web-terminal-e2e/1283666065389260800/artifacts/openshift-web-terminal-e2e/pods/

This logs are collected from the current job.

File: .cico/cico_openshift_e2e.sh [hunk 2]
```diff
@@ -13,6 +13,11 @@ set -ex
 export CI="openshift"
 export ARTIFACTS_DIR="/tmp/artifacts"
 
+# Component is defined in Openshift CI job configuration. See: https://github.com/openshift/release/blob/master/ci-operator/config/devfile/devworkspace-operator/devfile-devworkspace-operator-master__v4.yaml#L8
+export CI_COMPONENT="devworkspace-operator"
+# OPENSHIFT_BUILD_NAMESPACE env var exposed by Openshift CI. More info about how images are builded in Openshift CI: https://github.com/openshift/ci-tools/blob/master/TEMPLATES.md#parameters-available-to-templates
+export CI_CONTROLLER_IMAGE=registry.svc.ci.openshift.org/${OPENSHIFT_BUILD_NAMESPACE}/stable:${CI_COMPONENT}
```
[the remark this answers is not in the document]
Comment (flacatus) [reply .cico/cico_openshift_e2e.sh:19]: I tried but is replaced here with default image: https://github.com/devfile/devworkspace-operator/blob/master/Makefile#L104

File: Makefile [hunk 3]
```diff
@@ -19,6 +19,7 @@ _print_vars:
 	@echo "Current env vars:"
 	@echo "    NAMESPACE=$(NAMESPACE)"
 	@echo "    IMG=$(IMG)"
+	@echo "    CI_CONTROLLER_IMAGE=$(CI_CONTROLLER_IMAGE)"
```
Comment (sleshchenko) [inline Makefile]:
```suggestion
```

File: test/e2e/pkg/tests/cloud_shell_tests.go [hunk 4]
```diff
@@ -20,24 +20,25 @@ import (
 	"github.com/onsi/gomega"
 )
 
-var _ = ginkgo.Describe("[Create Cloud Shell Workspace]", func() {
-	ginkgo.It("Add cloud shell to cluster", func() {
-		label := "controller.devfile.io/workspace_name=cloud-shell"
+var _ = ginkgo.Describe("[Create Openshift Web Terminal Workspace]", func() {
```
Comment (sleshchenko) [inline test/e2e/pkg/tests/cloud_shell_tests.go:23]:
```suggestion
var _ = ginkgo.Describe("[Create OpenShift Web Terminal Workspace]", func() {
```

File: Makefile (hunk 3 above)
Comment (flacatus) [reply Makefile]: Just pushed now :D

**Pull request merged**